app/controllers/role_lists_controller.rb

```diff
@@ -91,18 +93,15 @@ def group
   end
 
   def people
-    @people ||= Person.where(id: people_ids).uniq
-  end
-
-  def people_ids
-    list_param(:ids)
-  end
-
-  def person_filter
-    @person_filter ||= Person::Filter::List.new(@group, current_user, list_filter_args)
+    @people ||= all_filtered_or_listed_people
   end
 
   def role_list
-    @role_list ||= Role::List.new(current_ability, params)
+    @role_list ||= begin
+      role_list_params = params.dup
+      role_list_params[:ids] = all_filtered_or_listed_people.unscope(:order).pluck(:id).join(",")
```

Comment:
Won't this potentially result in hundreds or thousands of ids when the `ids=all` param is set?

Reply:
yes. the proper solution would be to push the "all + filter-conditions" further down or moving the mutation into a background-job. I opted against that edge-case of changing 1000 roles this way to save a time. If people are using this to mutate (add/remove/move) many roles at once (unlikely to me) AND it becomes a problem (very likely), we can fix it then.